lib/logger/src/main/scala/org.bitbucket.pshirshov.izumi/logger/LogSink.scala

```diff
@@ -0,0 +1,38 @@
+package org.bitbucket.pshirshov.izumi.logger
+
+import scala.annotation.tailrec
+
+trait LogSink {
+
+  def messageParser(msg: Log.Message): Map[String, Any] = {
```

Comment:
Parsing should happen BEFORE sink.

Logger message should have the following signature: `(args: Map[String, Any], template: String, message: String)` where message is a rendered message, template is original message with placeholders instead of arguments. Sink should accept these messages.

Your parsing should happen withing your Logger implementation.